pkg/executables/clusterctl.go

```diff
@@ -155,19 +155,40 @@ func (c *Clusterctl) writeInfrastructureBundle(clusterSpec *cluster.Spec, rootFo
 // BackupManagement save CAPI resources of a workload cluster before moving it to the bootstrap cluster during upgrade.
 func (c *Clusterctl) BackupManagement(ctx context.Context, cluster *types.Cluster, managementStatePath string) error {
 	filePath := filepath.Join(".", cluster.Name, managementStatePath)
-	err := os.MkdirAll(filePath, os.ModePerm)
-	if err != nil {
-		return fmt.Errorf("could not create backup file for CAPI objects: %v", err)
-	}
 
-	_, err = c.Execute(
-		ctx, "move",
-		"--to-directory", filePath,
-		"--kubeconfig", cluster.KubeconfigFile,
-		"--namespace", constants.EksaSystemNamespace,
-	)
-	if err != nil {
-		return fmt.Errorf("failed taking backup of CAPI objects: %v", err)
+	// check for existing backup to prevent partial overwrites
+	_, err := os.Stat(filePath)
+	if err == nil {
+		tempPath := fmt.Sprintf("%s_Temp", managementStatePath)
+		err = c.BackupManagement(ctx, cluster, tempPath)
```

Comment:
Can we avoid this recursive call (if I'm reading this correctly)? Perhaps we can extract the backup logic to another method and then passing the path as an argument to it.